### DISCUSSION TITLE
Kurtosis and Tilt

### DISCUSSION THREAD

**Discussion by CyrusVorwald** [kurtosis-tech/kurtosis#1065] [Q&A] [answered]
Hello,

I am wondering what the differences are between Kurtosis and Tilt and whether they can be used in conjunction with each other or if they serve the same purposes. I don't have any specific workflows in mind, although this did come from observing that [Wormhole](https://github.com/wormhole-foundation/wormhole/tree/main) uses Tilt for its devnet and some of the networks in that, such as NEAR, use Kurtosis for its local network container.

**Comment by leeederek** [comment, marked as answer]
Hey @CyrusVorwald  - great question. Kurtosis and Tilt do very similar things for a developer but Kurtosis is not meant to be used alongside or with Tilt. **TLDR: Kurtosis can be used for dev, test, and prod while Tilt is a tool for dev only.**

To answer your question directly about the differences, the main one I want to call out is actually from Tilt.dev's own Github: `Kubernetes for prod, tilt for dev`. Kurtosis is a tool that is meant for dev, testing and production - we believe DevOps needs to evolve & a single tool across the development lifecycle will greatly simplify how devs write software by removing the pain points that come with switching between different tools and frameworks at each step of the development process. To be more specific, here are a few differences i'll call out:
* **Kurtosis began and still excels at testing while tilt is aimed at dev.** Environments brought up by Kurtosis are ephemeral by default, designed to be reproducible and cheap, and work over Docker or Kubernetes. Kurtosis environment definitions are also composable and have specific features/guard-rails that gracefully handle sequential set up logic & the passing of runtime dependencies between services - features that our users find quite valuable when it comes to performing complex end-to-end or integration tests on blockchain infrastructure (e.g. handling of genesis files, sequential steps required to bootstrap a node and connect it to a network, network behaviors/changes, etc)
* **Kurtosis will *very soon* support production deploy out-of-the-box** (<1 month away from a real customer trying it out). Write your environment definition once, click a button to deploy it on remote, managed infrastructure. This same mechanism can be used to deploy persistent development or staging environments for your teams to collaborate and dev on the backend you're familiar with, at any scale you need, and with complete reproducibility. And this ensures that the promotion from dev-->test-->prod is a button click.
* **Kurtosis works on both Kubernetes and Docker.** Tilt only works on Kubernetes. This means a user or project using Tilt needs to already have the infrastructure and in-house expertise to manage + provision k8s and understand the k8s-specific primitives for their application.

For things in Tilt that you may be looking for in Kurtosis:
* We've shipped a v0 frontend for developers to interact, build, and deploy their systems. Locally, this is going to be `localhost:9711` when you build an environment using Kurtosis. Our web-app is very close to being publicly available as well - we're targeting end of month alongside our cloud offering Alpha.
* [Kurtosis idempotent runs](https://docs.kurtosis.com/explanations/how-do-idempotent-runs-work/) was shipped quite recently - update your environment definition and Kurtosis will re-apply the differences to your environment, enabling dev workflows. This works right now and you can give it a spin if you'd like! We're here to help if you need a hand.

I hope this answers your question!

**Reply by CyrusVorwald** [reply]
Thank you for your prompt and detailed response!
